pkg/archive/archive_windows.go

```diff
@@ -42,11 +42,14 @@ func CanonicalTarNameForPath(p string) (string, error) {
 // chmodTarEntry is used to adjust the file permissions used in tar header based
 // on the platform the archival is done.
 func chmodTarEntry(perm os.FileMode) os.FileMode {
-	perm &= 0755
+	//perm &= 0755 // this 0-ed out tar flags (like link, regular file, directory marker etc.)
+	permPart := perm & os.ModePerm
```

Comment:
It is for readability: make it very explicits that the bits are splitted between a permission block, and the other bits.
As os.ModePerm and the other litterals used in bitwise computations here are compile-time constant, I assume all of this will be optimized away, and opted for clarity.